client/js/users.js

```diff
@@ -0,0 +1,161 @@
+    const userPost = document.querySelector("#usersResults");
+    userPost.innerHTML = ""
+    for (let i = 0; i < userId.length; i++) {
+        let response = await axios.get(`http://localhost:11000/users/${userId[i]}`)
+        response.data.body.forEach((users) => {
+            let searchDiv = document.createElement('div')
+            searchDiv.className = "searchResultsDiv"
+            userPost.append(searchDiv)
+            let listPost = document.createElement('li')
+            let listPost2 = document.createElement('li')
+            listPost.innerText = `Name:${users.firstname} ${users.lastname}`;
+            listPost2.innerText = `Age: ${users.age}`;
+            searchDiv.append(listPost, listPost2)
+        })
+    }
+}
+
+
+async function searchTopLeftUser() {
+    event.preventDefault();
+    let userName = document.querySelector('#search2').value;
+    const findAllUsers = await axios.get(`http://localhost:11000/users/`)
+    let responsedata = findAllUsers.data.body
+    let userId = searchNames(responsedata, userName)
+    const title = document.getElementById("allTheUsers")
+    title.innerHTML = `Results for ${userName}`;
+    const userPost = document.querySelector("#usersResults");
+    userPost.innerHTML = ""
+    for (let i = 0; i < userId.length; i++) {
+        let response = await axios.get(`http://localhost:11000/users/${userId[i]}`)
+        response.data.body.forEach((users) => {
+            let searchDiv = document.createElement('div')
+            searchDiv.className = "searchResultsDiv"
+            userPost.append(searchDiv)
+            let listPost = document.createElement('li')
+            let listPost2 = document.createElement('li')
+            listPost.innerText = `Name:${users.firstname} ${users.lastname}`;
+            listPost2.innerText = `Age: ${users.age}`;
+            searchDiv.append(listPost, listPost2)
+        })
+    }
+}
+
+const searchNames = (data, userName) => {
+    let id = []
+    for (result of data) {
+        if (userName.toUpperCase() === result.firstname.toUpperCase()) {
+            id.push(result.user_id)
+        } else if (userName.toUpperCase() === result.lastname.toUpperCase()) {
+            id.push(result.user_id)
+        } else if (userName.toUpperCase() === result.firstname.toUpperCase() + " " + result.lastname.toUpperCase()) {
+            id.push(result.user_id)
+        }
+    }
+    return id
+}
+
+const addAUser = async () => {
+    event.preventDefault();
+    const firstname = document.querySelector("#searchBoxAddFN").value
+    const lastname = document.querySelector("#searchBoxAddLN").value
+    const age = document.querySelector("#searchBoxAddAge").value
+    const findAllUsers = await axios.get(`http://localhost:11000/users/`)
+    let responseData = findAllUsers.data.body
+    let id = filerResults(responseData, firstname, lastname, age)
+    if (id === -1) {
+        await axios.post(`http://localhost:11000/users/`, { firstname, lastname, age });
```

Comment:
Two things here:
1- the server route actually needs to be fixed. Users.js in the server, line 91:
router.post("/**:id**", addUser); // add a user
the **:id** needs to be removed. Better you do it along with any other changes and include the file into your next commit here.

2- the format of the body argument youre passing in, the { firstname, lastname, age } work because those are already variables. but if you use values instead like { "Kathy", "Puma", 23 } you'll get errors. better to use the format of `key: value/variable` all the time.
```suggestion
        await axios.post(`http://localhost:11000/users/`, { firstname: firstname, lastname: lastname, age: age });
```